app/routes/synonyms/synonyms.server.ts

```diff
@@ -334,16 +334,29 @@ export async function deleteSynonyms(synonymId: string) {
     },
   })
 
-  const writePromises = results.Items.map(async ({ eventId }) => ({
-    PutRequest: {
-      Item: {
-        synonymId: crypto.randomUUID(),
-        eventId,
-        slug: slug(eventId),
-        initialDate: await getOldestDate(eventId),
-      },
-    },
-  }))
+  const writePromises = results.Items.map(async ({ eventId }) => {
+    const initialDate = await getOldestDate(eventId)
```

Comment:
This makes the deletion operation accidentally quadratic in the number of synonyms.

Reply:
Could you explain this a bit more? I'm not sure what you mean. The results will only contain unique eventIds since that is the partition key for the synonyms table, so there will only ever be 1 per eventId. So there would be no duplicates in results. the getOldestDate function and the function it uses to accomplish its job, getSynonymMembers, are not table scans. So I'm unsure how this could be accidentally quadratic.

Reply:
It's quadratic in the sense that it is O(number of synonyms x number of Circulars per synonym). Do you agree?

Reply:
This is how I was thinking about it. This is how I have it in this PR:
![synonym_delete_as_is](https://github.com/user-attachments/assets/dfcb0bd2-627e-4439-b77b-41bd4dba3cbe)

If I query to see if there are circulars with that eventId and then only call the getOldestDate function if it's not empty, in most cases it duplicates the call to getSynonymMembers.
![counting_circulars_first](https://github.com/user-attachments/assets/0c4d4984-515d-47ca-84c9-eb32d5224ffe)